src/polyfem/state/StateDiff.cpp

```diff
@@ -438,7 +450,7 @@ namespace polyfem
 					Eigen::VectorXd b_ = rhs_;
 					b_(boundary_nodes).setZero();
 
-					auto solver = polysolve::linear::Solver::create(args["solver"]["linear"], adjoint_logger());
+					auto solver = polysolve::linear::Solver::create(args["solver"]["adjoint_linear"], adjoint_logger());
```

Comment:
Yes, it is. I put it here because it's a choice per state, it may differ for different states. I can put it under `opt_args["states"]["adjoint_linear"]` otherwise.

Reply:
nah it is ok